### PR TITLE
Unique pallet with merge conflicts

### PR DESCRIPTION
**Pull request by rohibhan**
Created a new branch - cherry picked all previous commits related to unique into this new branch mentioned in title.

### CODE REVIEW

File: pallets/unique/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+[package]
+name = 'pallet-unique'
+description = 'Manage Unique Entries.'
+version = '0.8.0'
```
Comment (amarts) [inline pallets/unique/Cargo.toml]:
```suggestion
version = '0.8.1'
```

**Review by amarts — COMMENTED** [review]
Don't we need anything in runtime to see this pallet in explorer app?

File: pallets/unique/Cargo.toml (hunk 1 above)
Comment (rohibhan) [reply pallets/unique/Cargo.toml]: version changed

**Comment by rohibhan** [comment]
> Don't we need anything in runtime to see this pallet in explorer app?

Added runtime.

**Comment by amarts** [comment]
```
 error[E0599]: no function or associated item named `benchmarks` found for struct `pallet_unique::Pallet` in the current scope
      --> /home/runner/work/cord/cord/runtime/src/lib.rs:1450:4
       |
  1450 |             list_benchmarks!(list, extra);
       |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `Pallet<Runtime>`
       |
       = note: this error originates in the macro `$crate::list_benchmark` which comes from the expansion of the macro `list_benchmarks` (in Nightly builds, run with -Z macro-backtrace for more info)

  error[E0599]: no function or associated item named `run_benchmark` found for struct `pallet_unique::Pallet` in the current scope
      --> /home/runner/work/cord/cord/runtime/src/lib.rs:1477:4
       |
  1140 |           [pallet_unique, Unique]
       |  _________________________-
  1141 | |         [pallet_registry, Registry]
  1142 | |         [pallet_did, Did]
  1143 | |         [pallet_did_names, DidNames]
  ...    |
  1477 | |             add_benchmarks!(params, batches);
       | |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `Pallet<Runtime>`
  ...    |
       |
       = note: this error originates in the macro `$crate::add_benchmark` which comes from the expansion of the macro `add_benchmarks` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by rohibhan** [comment]
> ```
>  error[E0599]: no function or associated item named `benchmarks` found for struct `pallet_unique::Pallet` in the current scope
>       --> /home/runner/work/cord/cord/runtime/src/lib.rs:1450:4
>        |
>   1450 |             list_benchmarks!(list, extra);
>        |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `Pallet<Runtime>`
>        |
>        = note: this error originates in the macro `$crate::list_benchmark` which comes from the expansion of the macro `list_benchmarks` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
>   error[E0599]: no function or associated item named `run_benchmark` found for struct `pallet_unique::Pallet` in the current scope
>       --> /home/runner/work/cord/cord/runtime/src/lib.rs:1477:4
>        |
>   1140 |           [pallet_unique, Unique]
>        |  _________________________-
>   1141 | |         [pallet_registry, Registry]
>   1142 | |         [pallet_did, Did]
>   1143 | |         [pallet_did_names, DidNames]
>   ...    |
>   1477 | |             add_benchmarks!(params, batches);
>        | |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `Pallet<Runtime>`
>   ...    |
>        |
>        = note: this error originates in the macro `$crate::add_benchmark` which comes from the expansion of the macro `add_benchmarks` (in Nightly builds, run with -Z macro-backtrace for more info)
> ```

Working on this. I think I need benchmarking for unique is it?

**Comment by amarts** [comment]
Can we add a **optional** - 'reference' field in the extrinsic (and should be stored, and query-able) in unique?

For example

I send 'amar-dhiway' as reference, along with the extrinsic, and I should be able to fetch back the details of identifier, hash and status from querying 'amar-dhiway' string. 

Now, think this can be an URL too, which means we can check if a URL is having same hash as it was anchored for etc... now helping any verification on url possible.

It can be filename if its a binary file.

or `None` if user doesn't want anything.

**Comment by smohan-dw** [comment]
> 



> Can we add a **optional** - 'reference' field in the extrinsic (and should be stored, and query-able) in unique?
> 
> For example
> 
> I send 'amar-dhiway' as reference, along with the extrinsic, and I should be able to fetch back the details of identifier, hash and status from querying 'amar-dhiway' string.
> 
> Now, think this can be an URL too, which means we can check if a URL is having same hash as it was anchored for etc... now helping any verification on url possible.
> 
> It can be filename if its a binary file.
> 
> or `None` if user doesn't want anything.

We need to ensure the uniqueness of the reference.

**Comment by amarts** [comment]
@rohibhan @roynrishingha Can one of you try to rebase this on top of #223  and see if there is any major conflicts? (Don't push that immediately).

I prefer to merge it into `develop` sooner so we can quickly get everyone to compile and build the new code to understand how the membership and transaction-fee less pallet code looks like.

**Comment by rohibhan** [comment]
> @rohibhan @roynrishingha Can one of you try to rebase this on top of #223 and see if there is any major conflicts? (Don't push that immediately).
> 
> I prefer to merge it into `develop` sooner so we can quickly get everyone to compile and build the new code to understand how the membership and transaction-fee less pallet code looks like.

Working on it

**Comment by roynrishingha** [comment]
@amarts as soon as I'm done with benchmarking I'll rebase and resolve conflicts and make this PR ready to review.

**Comment by roynrishingha** [comment]
Due to huge merge conflicts this PR is abandoned and the feature is added using PR #229

**Pull request closed without merging**